gnosis/safe/api/transaction_service_api/transaction_service_api.py

```diff
@@ -114,6 +114,34 @@ def parse_signatures(cls, raw_tx: Dict[str, Any]) -> Optional[bytes]:
                 ]
             )
 
+    def _build_transaction_service_tx(
+        self, tx_raw: Dict[str, Any]
+    ) -> TransactionServiceTx:
+        signatures = self.parse_signatures(tx_raw)
+        safe_tx = TransactionServiceTx(
+            tx_raw["proposer"],
+            self.ethereum_client,
+            tx_raw["safe"],
+            tx_raw["to"],
+            int(tx_raw["value"]),
+            HexBytes(tx_raw["data"]) if tx_raw["data"] else b"",
+            int(tx_raw["operation"]),
+            int(tx_raw["safeTxGas"]),
+            int(tx_raw["baseGas"]),
+            int(tx_raw["gasPrice"]),
+            tx_raw["gasToken"],
+            tx_raw["refundReceiver"],
+            signatures=signatures if signatures else b"",
+            safe_nonce=int(tx_raw["nonce"]),
+            chain_id=self.network.value,
+        )
+        tx_hash = (
```

Comment:
You should do the check here to raise the exception, so any endpoint where this method is called raises